extendr-api/src/robj/into_robj.rs

```diff
@@ -587,28 +570,37 @@ macro_rules! impl_from_as_iterator {
 //     }
 // }
 
-// Template constants are still unstable in rust.
-impl_from_iter! {[T; 1]}
-impl_from_iter! {[T; 2]}
-impl_from_iter! {[T; 3]}
-impl_from_iter! {[T; 4]}
-impl_from_iter! {[T; 5]}
-impl_from_iter! {[T; 6]}
-impl_from_iter! {[T; 7]}
-impl_from_iter! {[T; 8]}
-impl_from_iter! {[T; 9]}
-impl_from_iter! {[T; 10]}
-impl_from_iter! {[T; 11]}
-impl_from_iter! {[T; 12]}
-impl_from_iter! {[T; 13]}
-impl_from_iter! {[T; 14]}
-impl_from_iter! {[T; 15]}
-impl_from_iter! {[T; 16]}
-impl_from_iter! {[T; 17]}
-impl_from_iter! {[T; 18]}
-impl_from_iter! {[T; 19]}
-impl_from_iter! {Vec<T>}
-impl_from_iter! {&Vec<T>}
+// TODO: Is this still relevant?
+// We would love to do a blanket IntoIterator impl.
+// But the matching rules would clash with the above.
+
+impl<'a, T, const N: usize> From<[T; N]> for Robj
+where
+    Self: 'a,
+    T: ToVectorValue,
+{
+    fn from(val: [T; N]) -> Self {
+        val.into_iter().collect_robj()
+    }
+}
+
+impl<T> From<Vec<T>> for Robj
+where
+    T: ToVectorValue,
+{
+    fn from(value: Vec<T>) -> Self {
+        value.into_iter().collect_robj()
+    }
+}
+
+impl<T> From<&Vec<T>> for Robj
+where
+    T: ToVectorValue + Clone,
+{
+    fn from(value: &Vec<T>) -> Self {
+        value.into_iter().cloned().collect_robj()
```

Comment:
Do we not have an impl for borrow vector values? ie can't you remove `cloned()` here?

Reply:
Yes I get that you have references, I'm just saying that you can directly convert references to Robj:

```rust
>> test!{
   let vector = vec![1, 2, 3].into_iter().collect_robj();
   call!("print", pairlist!(vector));
}
[[1]]
[1] 1 2 3

()
```

Reply:
Okay. I've made it compile. But I do not trust that this works at all. I will see if there are tests, but it is simply impossible for me to see how `ToVectorValue` is defined for ref-types in a non-clone way.

Reply:
Fine, I believe it works:

```rust
    #[test]
    fn test_to_robj_from_references() {
        let vs = vec![0_f32, 2.4, 2.3, 222.2];
        test! { Robj::from(&vs); }
    }
```
This compiled and ran fine..

Reply:
```rust
    #[test]
    fn test_to_robj_from_references() {
        let vs = vec![0_f32, 2.4, 2.3, 222.2];
        test! { let vs_back = Robj::from(&vs); println!("{vs_back:?}"); }
    }
```
`[0.0, 2.4000000953674316, 2.299999952316284, 222.1999969482422]`